libraries/src/AWS.Lambda.Powertools.Common/Core/SystemWrapper.cs

```diff
@@ -78,4 +83,51 @@ public double GetRandom()
     {
         return new Random().NextDouble();
     }
+
+    /// <inheritdoc />
+    public void SetEnvironmentVariable(string variable, string value)
+    {
+        _powertoolsEnvironment.SetEnvironmentVariable(variable, value);
+    }
+
+    /// <inheritdoc />
+    public void SetExecutionEnvironment<T>(T type)
+    {
+        const string envName = Constants.AwsExecutionEnvironmentVariableName;
+
+        var envValue = new StringBuilder();
+
+        // If there is an existing execution environment variable add the annotations package as a suffix.
+        if(!string.IsNullOrEmpty(GetEnvironmentVariable(envName)))
+        {
+            envValue.Append($"{GetEnvironmentVariable(envName)} ");
+        }
+
+        var assemblyVersion = _powertoolsEnvironment.GetAssemblyVersion(type);
+        var assemblyName =_powertoolsEnvironment.GetAssemblyName(type);
+
+        envValue.Append($"{ParseAssemblyName(assemblyName)}/{assemblyVersion}");
```

Comment:
For defensive purposes can you check if the value of `ParseAssemblyName(assemblyName)` is already in the environment variable? I want to prevent a potential issue if there is ever a bug where the `SetExecutionEnvironment` method is called per request triggering the environment variable to keep growing and growing.

Reply:
Added verification for duplicates. If the parsed assembly name is already present in the environment variables it will not be added again.